src/packsquash.js

```diff
@@ -0,0 +1,59 @@
+import { endGroup, exportVariable, getBooleanInput, startGroup, warning } from '@actions/core';
+import { exec } from '@actions/exec';
+import { rmSync } from 'fs';
+import { addProblemMatcher, removeProblemMatcher } from './problem-matcher';
+import { Options } from './options';
+
+/**
+ * @param {WorkingDirectory} workingDirectory
+ * @returns {Promise<void>}
+ */
+export async function printPackSquashVersion(workingDirectory) {
+    startGroup('PackSquash version');
+    await exec(workingDirectory.packsquash, ['--version']);
+    endGroup();
+}
+
+/**
+ * @param {WorkingDirectory} workingDirectory
+ * @returns {Promise<void>}
+ */
+export async function runPackSquash(workingDirectory) {
+    async function run(description) {
+        addProblemMatcher(workingDirectory.problemMatcher);
+        if (description) {
+            startGroup(`PackSquash output (${description})`);
+        } else {
+            startGroup('PackSquash output');
+        }
+        const exitCode = await exec(workingDirectory.packsquash, [workingDirectory.options]);
+        endGroup();
+        removeProblemMatcher(workingDirectory.problemMatcher);
+        return exitCode;
+    }
+
+    let exitCode = await run();
+    switch (exitCode) {
+        case 0:
+            break;
+        case 129:
+            warning('PackSquash reported that the previous ZIP file could not be used to speed up processing. Discarding it.');
+            rmSync(workingDirectory.zip);
+            exitCode = await run('discarded previous ZIP file');
+            if (exitCode !== 0) {
+                process.exit(exitCode);
+            }
+            break;
+        default:
+            // Any other PackSquash error
+            process.exit(exitCode);
```

Comment:
Did I say already how much I like throwing exceptions? :wink: 

```suggestion
            if (exitCode !== 0) {
                throw new Error(`PackSquash finished with an error code: ${exitCode}`);
            }
            break;
        default:
            // Any other PackSquash error
            throw new Error(`PackSquash finished with an error code: ${exitCode}`);
```

Reply:
Fix by a1fe71f1a1d5edaae667fea40e3ff6957eb7eebf